website/docs/cdktf/concepts/resources.mdx

```diff
@@ -370,6 +370,78 @@ import imports.kubernetes.namespace.NamespaceMetadata;
 
 </CodeTabs>
 
+## Refactoring & Renaming Resources
+
+When working with your infrastructure definitions and the need arises to refactor or rename resources without destroying and recreating them, you can leverage the `moveTo` function. This section outlines how to perform resource moves, rename resources, and enable foreach on existing resources in your CDKTF project.
+
+### Performing Resource Moves
+
+The `moveTo` function is available on all resources and is used for relocating a resource to the location specified by the string target `moveTarget`. To set the `moveTarget` for a resource, utilize the `addMoveTarget` function that is present on the resource to move to.
+
+```ts
+new S3Bucket(this, "test-bucket-move-to", {
+  bucket: "move-bucket-name",
+}).addMoveTarget("move-s3");
+
+new S3Bucket(this, "test-bucket-move-from", {
+  bucket: "move-bucket-name",
+}).moveTo("move-s3");
+```
+
+After deployment, both the resource being moved and the move target on the destination resource can be removed.
+
+#### Move Targets
+
+A `moveTarget` is accessible anywhere within the context of the stack where it is created. Whether it's set at the root of the stack or within a nested construct, you can access it accordingly.
+
+```ts
+new S3Bucket(this, "test-bucket-1", {
+  bucket: "test-move-bucket-name-1",
+}).moveTo("move");
+
+// Contains resource with moveTarget "move"
+new NestedConstructToMoveTo(this, "construct-to-move-to");
+```
+
+### Renaming a Resource
+
+To rename a resource, you can simply use the `renameResourceId` function available on the resource.
+
+```ts
+new S3Bucket(this, "bucket-id", {
+  bucket: "bucket-rename",
+}).renameResourceId("bucket-new-id");
+```
+
+After deploying the resource, you can update your resource to use the newly assigned resource ID, and the `renameResourceId` call can be safely removed.
+
+### Enabling `foreach` on a Resource
+
+If you wish to incorporate a deployed resource into a composition using `foreach`, you can achieve this without destroying the resource by specifying the desired index using the `moveTo` function. The index should correspond to the key found in the `TerraformIterator` named iterator
```

Comment:
```suggestion
To incorporate a deployed resource into a composition without destroying the resource, use the `foreach` function and specify an index for the `moveTo` function. The index should correspond to the key in the `TerraformIterator` named `iterator`. 
```
Is that clearer?

Reply:
Far clearer, changed your suggestion to use some more correct wording but otherwise [included](https://github.com/hashicorp/terraform-cdk/pull/3152/commits/70e4146b79b73f51e3b568df7fd5918366cb5b49) 👍